### PR TITLE
Async eth.get_balance, eth.get_code, eth.get_transaction_count

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?

Added async `eth.get_balance`, `eth.get_code`, and `eth.get_transaction_count` (all methods that use the `block_id_munger`). 

Related to Issue #1413 

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)
- [x] Manual Testing

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: web3/method.py [hunk 1]
```diff
@@ -40,8 +40,6 @@
     from web3 import Web3  # noqa: F401
     from web3.module import Module  # noqa: F401
 
-Munger = Callable[..., Any]
```
Comment (fselmo) [inline web3/method.py:43]: Is this change just to simplify it a bit since this is pretty general anyhow?

**Review by fselmo — APPROVED** [review]
@kclowes LGTM!

File: web3/method.py (hunk 1 above)
Comment (kclowes) [reply web3/method.py:43]: oops, I don't remember why I did this. Will put it back!

**Comment by minhvutnt** [comment]
my code w3.eth.get_code(account.address)) return b'' (empty)
with account.address = 0xBeddf5c6d76B5eb2c615c31d9aeC4fE4d334380e
What happend?

**Pull request merged**